agent/util/archive.go

```diff
@@ -40,84 +43,73 @@ func FindContentsToArchive(ctx context.Context, rootPath string, includes, exclu
 	return out, nil
 }
 
-func streamArchiveContents(ctx context.Context, rootPath string, includes, excludes []string) <-chan ArchiveContentFile {
-	outputChan := make(chan ArchiveContentFile)
+func streamArchiveContents(ctx context.Context, rootPath string, includes, excludes []string) ([]ArchiveContentFile, error) {
+	archiveContents := []ArchiveContentFile{}
+	catcher := grip.NewCatcher()
 
-	go func() {
-		defer close(outputChan)
-		defer func() {
-			panicErr := recovery.HandlePanicWithError(recover(), nil,
-				"streaming archive contents")
-			if panicErr != nil {
-				outputChan <- ArchiveContentFile{err: panicErr}
-			}
-		}()
-		for _, includePattern := range includes {
-			dir, filematch := filepath.Split(includePattern)
-			dir = filepath.Join(rootPath, dir)
+	for _, includePattern := range includes {
+		dir, filematch := filepath.Split(includePattern)
+		dir = filepath.Join(rootPath, dir)
 
-			if !utility.FileExists(dir) {
-				continue
-			}
+		if !utility.FileExists(dir) {
+			continue
+		}
 
-			if ctx.Err() != nil {
-				outputChan <- ArchiveContentFile{err: errors.New("archive creation operation canceled")}
-				return
-			}
+		if ctx.Err() != nil {
+			return nil, errors.New("archive creation operation canceled")
+		}
 
-			var walk filepath.WalkFunc
+		var walk filepath.WalkFunc
 
-			if filematch == "**" {
-				walk = func(path string, info os.FileInfo, err error) error {
+		if filematch == "**" {
+			walk = func(path string, info os.FileInfo, err error) error {
+				for _, ignore := range excludes {
+					if match, _ := filepath.Match(ignore, path); match {
+						return nil
+					}
+				}
+
+				archiveContents = append(archiveContents, ArchiveContentFile{path, info, nil})
+				return nil
+			}
+			_ = filepath.Walk(dir, walk)
+		} else if strings.Contains(filematch, "**") {
+			globSuffix := filematch[2:]
+			walk = func(path string, info os.FileInfo, err error) error {
+				if strings.HasSuffix(filepath.Base(path), globSuffix) {
 					for _, ignore := range excludes {
 						if match, _ := filepath.Match(ignore, path); match {
 							return nil
 						}
 					}
 
-					outputChan <- ArchiveContentFile{path, info, nil}
-					return nil
+					archiveContents = append(archiveContents, ArchiveContentFile{path, info, nil})
 				}
-				_ = filepath.Walk(dir, walk)
-			} else if strings.Contains(filematch, "**") {
-				globSuffix := filematch[2:]
-				walk = func(path string, info os.FileInfo, err error) error {
-					if strings.HasSuffix(filepath.Base(path), globSuffix) {
+				return nil
+			}
+			_ = filepath.Walk(dir, walk)
```

Comment:
why aren't we catching this error or the one at 75?

Reply:
I totally missed that there were several `filepath.Walk`s. Thanks!